homeassistant/components/influxdb/__init__.py

```diff
@@ -156,6 +156,28 @@ def get_influx_connection(client_kwargs, bucket):
     return influx
 
 
+def create_influx_url(conf):
```

Comment:
This can be a custom config schema validator.

Reply:
Do you mean change it so it can be used like this:
```python
{
    DOMAIN: vol.All(
        _CONFIG_SCHEMA.extend(COMPONENT_CONFIG_SCHEMA_CONNECTION),
        create_influx_url,
    )
}
```
So it creates the influx URL as part of validation of the config basically?  That makes sense but still not sure how to handle the sensor then.  In the sensor, `PLATFORM_SCHEMA` gets initialized like this:
``` python
PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(COMPONENT_CONFIG_SCHEMA_CONNECTION).extend(
    {
        vol.Exclusive(CONF_QUERIES, "queries"): [_QUERY_SCHEMA["InfluxQL"]],
        vol.Exclusive(CONF_QUERIES_FLUX, "queries"): [_QUERY_SCHEMA["Flux"]],
    }
)
```
If I try and use `vol.All` to introduce a validator function then it'll break on startup when core tries to extend the base sensor config with this PLATFORM_SCHEMA since its not a simple schema anymore.

Reply:
Why can't we do something like this?
https://github.com/home-assistant/core/blob/4170eb0f83ee5fef11eef1b656d3aa4696c67b95/homeassistant/components/azure_service_bus/notify.py#L31-L44

Reply:
Huh.  That looks perfect but I thought that was what I tried and had trouble with.  I'll try it again though, that definitely looks like the best option.  With that I could do any additional validation required in a custom validator function.

Reply:
I'm not really sure what issue I was encountering before, using `vol.All` had no issues with helper functions had no issues this time. I'll chalk it up to me misunderstanding some other issue and drawing the wrong conclusions.

`create_influx_url` is now a helper fuction and working as you described 👍